### PR TITLE
Wait for nodes after CCM install in ci-entrypoint

### LINKED ISSUE #2921: ci-entrypoint.sh fails intermittently with "error: no objects passed to create" due to missing kubeadm-config ConfigMap
/kind bug
/kind flake

[Before submitting an issue, have you checked the [Troubleshooting Guide](https://capz.sigs.k8s.io/topics/troubleshooting.html)?]

**What steps did you take and what happened:**
The ci-entrypoint pre-submit CI job has been flaky lately and I've noticed a number of failing runs hit the same error here:
```
Error from server (NotFound): configmaps "kubeadm-config" not found
error: no objects passed to create
```
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/e0d9257431826beb489db1db65204f55ca8f2155/scripts/ci-entrypoint.sh#L149-L151 

I suspect this is some sort of race between kubeadm creating that ConfigMap and reading it here? And waiting for that to exist would fix this error?

Runs:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/2916/pull-cluster-api-provider-azure-ci-entrypoint/1601245720478748672
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/2752/pull-cluster-api-provider-azure-ci-entrypoint/1600913653387759616
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/2738/pull-cluster-api-provider-azure-ci-entrypoint/1599870510261342208
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/2411/pull-cluster-api-provider-azure-ci-entrypoint/1598784340660064256

**What did you expect to happen:**
The kubeadm-config ConfigMap exists before it's read so the test doesn't fail here.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

**What this PR does / why we need it**: This fixes an issue when testing external CCM (`TEST_CCM=true`) in ci-entrypoint.sh where we try to wait for nodes to be ready before CCM is installed (nodes won't be ready because cloud provider is not initialized). 

It also fails a race condition in the script which causes it to fail intermittently when kubeadm-config cm is not available right away.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2921 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
/hold

need to check something

**Comment by CecileRobertMichon** [comment]
/hold cancel

this fixes #2921 as well

File: scripts/ci-entrypoint.sh [hunk 1]
```diff
@@ -207,6 +211,9 @@ install_addons() {
             --set-string cloudControllerManager.clusterCIDR="${CCM_CLUSTER_CIDR}"
     fi
 
+    export -f wait_for_nodes
+    timeout --foreground 1800 bash -c wait_for_nodes
```
Comment (marosset) [inline scripts/ci-entrypoint.sh:215]: Are we worried that waiting for the CCM install will make it more likely we'll hit this timeout?
Given that the timeout is 30 mins I would hope not but I wanted to ask...

Comment (CecileRobertMichon) [reply scripts/ci-entrypoint.sh:215]: I don't think I understand the question, could you please expand?

**Comment by CecileRobertMichon** [comment]
> Pod got deleted unexpectedly 

maybe prow is having issues?

/retest

File: scripts/ci-entrypoint.sh [hunk 2]
```diff
@@ -141,9 +141,16 @@ install_addons() {
 
     # Copy the kubeadm configmap to the calico-system namespace. This is a workaround needed for the calico-node-windows daemonset to be able to run in the calico-system namespace.
     "${KUBECTL}" create ns calico-system
+    for (( i=1; i<=30; i++ ))
+    do
+        if eval "${KUBECTL}" get configmap kubeadm-config --namespace=kube-system; then
```
Comment (jackfrancis) [inline scripts/ci-entrypoint.sh:146]: do we have to do eval here? does a negative lookup not produce a non-zero exit code (we should be able to just if/else based on the exit code of the command)

Comment (CecileRobertMichon) [reply scripts/ci-entrypoint.sh]: Good call, changed it to use until

File: scripts/ci-entrypoint.sh (hunk 1 above)
Comment (CecileRobertMichon) [reply scripts/ci-entrypoint.sh:213]: I've verified this doesn't cause issues in local testing

**Review by jackfrancis — APPROVED** [review]
/lgtm
/approve

**Pull request merged**